source/extensions/filters/http/jwt_authn/extractor.cc

```diff
@@ -325,12 +325,9 @@ absl::string_view ExtractorImpl::extractJWT(absl::string_view value_str,
   if (starting == value_str.npos) {
     return value_str;
   }
+
   // There should be two dots (periods; 0x2e) inside the string, but we don't verify that here
-  auto ending = value_str.find_first_not_of(ConstantBase64UrlEncodingCharsPlusDot, starting);
-  if (ending == value_str.npos) { // Base64Url-encoded string occupies the rest of the line
-    return value_str.substr(starting);
-  }
-  return value_str.substr(starting, ending - starting);
+  return value_str.substr(starting);
```

Comment:
Is a runtime guard warranted for this change? https://github.com/envoyproxy/envoy/blob/main/CONTRIBUTING.md#runtime-guarding

> Generally as a community we try to guard [...] most user-visible non-config-guarded changes to protocol processing (for example additions or changes to HTTP headers or how HTTP is serialized out) for non-alpha features. Feel free to tag @envoyproxy/maintainers if you aren't sure if a given change merits runtime guarding.

I'm slightly worried it will break runtime traffic for clients using it (https://www.hyrumslaw.com/). Though this is a very unique corner case and I doubt clients rely on this behavior. Either way, worth asking @envoyproxy/maintainers

Reply:
RFC 7519 implies that adding non-Base64URL characters after a valid JWT token makes it invalid. Thus, if a the JWT is followed by a space and some Base64URL characters or just followed by non Base64URL characters like ### (without any space) it should be considered invalid.